### DISCUSSION TITLE
### Summary

### DISCUSSION THREAD

**Discussion by srhise** [vercel/next.js#65292] [Help]
### Summary

In my NextJS application I have added google tag manager embed code like this in a separate component and then I call this component in _app.tsx, but when I add this my https://pagespeed.web.dev/ Total Blocking Time increased. I think google tag manager is loading asynchronously, but not me for me I think. Not sure what wrong I'm doing.

Can please some help? I'm using NextJS 13.3

### Additional information

```js
"use client"

import { GTM_ID, pageview } from "@/lib/gtm"
import { usePathname, useSearchParams } from "next/navigation"
import Script from "next/script"
import { useEffect } from "react"

export default function Analytics() {
  const pathname = usePathname()
  const searchParams = useSearchParams()

  useEffect(() => {
    if (pathname) {
      pageview(pathname)
    }
  }, [pathname, searchParams])

  if (process.env.NODE_ENV !== "production") {
    return null
  }

  return (
    <>
      <noscript>
        <iframe
          src={`https://www.googletagmanager.com/ns.html?id=${GTM_ID}`}
          height="0"
          width="0"
          style={{ display: "none", visibility: "hidden" }}
        />
      </noscript>
      <Script
        id="gtm-script"
        strategy="afterInteractive"
        dangerouslySetInnerHTML={{
          __html: `
    (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
    new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
    j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
    'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
    })(window,document,'script','dataLayer', '${GTM_ID}');
  `,
        }}
      />
    </>
  )
}
```

_Originally posted by @angrejkumar in https://github.com/vercel/next.js/discussions/59367_

**Comment by srhise** [comment]
This issue is also present in the `<GoogleTagManager/>` component. 

<img width="814" alt="image" src="https://github.com/vercel/next.js/assets/914113/79dff728-5201-496f-bcaf-b90edfaf9d09">

Page Speed Score: 

<img width="1056" alt="image" src="https://github.com/vercel/next.js/assets/914113/2f2c4f65-6c66-44b4-9c11-86884f3ea8f7">

When you look at the resulting HTML from the server, appears as if the script tag is simply added to the <head> with a preload attribute on the script tag, which is not what the docs say it should be doing.

<img width="706" alt="image" src="https://github.com/vercel/next.js/assets/914113/42b3b217-8337-45c2-be13-de720b68ad72">

**Comment by samcx** [comment]
@srhise Hi there, questions like this need to remain in <picture data-single-emoji=":nextjs:" title=":nextjs:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/nextjs/2a85d633ae594556.png" alt=":nextjs:" align="absmiddle"></picture> [discussions](https://github.com/vercel/next.js/discussions)!
